scripts/build_load_data.py

```diff
@@ -116,14 +116,19 @@ def max_consecutive_nans(ds):
                  keys=['total', 'consecutive', 'max_total_per_month'], axis=1)
 
 
-def copy_timeslice(load, cntry, start, stop, delta):
+def copy_timeslice(load, cntry, start, stop, delta, fn_load=None):
     start = pd.Timestamp(start)
     stop = pd.Timestamp(stop)
-    if start-delta in load.index and stop in load.index and cntry in load:
-        load.loc[start:stop, cntry] = load.loc[start-delta:stop-delta, cntry].values
+    if (start in load.index and stop in load.index):
+        if start-delta in load.index and stop-delta in load.index and cntry in load:
+            load.loc[start:stop, cntry] = load.loc[start-delta:stop-delta, cntry].values
+        elif fn_load is not None:
+            duration = pd.date_range(freq='h', start=start-delta, end=stop-delta)
+            load_raw = load_timeseries(fn_load, duration, [cntry], powerstatistics)
+            load.loc[start:stop, cntry] = load_raw.loc[start-delta:stop-delta, cntry].values
```

Comment:
Could this be shorter by reading in the time series segments anew in any case? Would be good to avoid nested if/else blocks. I haven't fully understood when the previous implementation fails. Could you give an example?

Reply:
Reading raw data takes a while, would try to omit whenever possible.

Breaks with current code for example for the year 2010, because whole January for GB is missing. The code tries to copy January from a previous/future year (I think future, but I'm not sure). However, those indices are not in `load`. Therefore, there's a length mismatch in the assignment: lhs has the length of January (744), rhs is empty.